### PR TITLE
add client for making tsa requests

### PR DESCRIPTION
**Pull request by bdehamer**
#### Summary
First step toward support for requesting timestamps from the [Timestamp Authority](https://github.com/sigstore/timestamp-authority).

This just adds a client for interacting w/ the TSA API. Plenty more work to do here to fully enable this feature.

### CODE REVIEW

File: src/external/tsa.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { ua } from '../util';
+import { checkStatus } from './error';
+
+export interface TimestampRequest {
+  artifact: string;
+  hashAlgorithm: string;
+  certificates?: boolean;
+  nonce?: number;
+  tsaPolicyOID?: string;
+}
```
Comment (bdehamer) [inline src/external/tsa.ts:26]: @malancas would love to get your eyes on this just to validate the "optionality" of the fields in the request.

Comment (bdehamer) [inline src/external/tsa.ts:26]: @malancas would love to get your eyes on this just to validate the "optionality" of the fields in the request.

Comment (malancas) [reply src/external/tsa.ts:26]: The optionality of these fields looks correct. Only the artifact and hash algorithm are required.

**Comment by bdehamer** [comment]
Closing until we work through some issues w/ the TSA API interface

**Pull request closed without merging**